.travis.yml

```diff
@@ -183,7 +183,7 @@ after_success:
   - if [ "$BUILD_DOCUMENTATION" = "ON" ] && [ "$TRAVIS_OS_NAME" == "linux" ]; then
       echo "------- Build Doxygen Documentation -------";
       pushd $TRAVIS_BUILD_DIR/build;
-      make docs;
+      make corenrn_docs;
```

Comment:
@alexsavulescu @nrnhines : should this be changed to avoid conflict? `docs` is much easier to remember / standard for any project.

Reply:
I greatly prefer leaving it as 'docs' and reverting this. The reason for it is that cmake raised an error when it was a submodule of nrn. Because nrn already has a target called 'docs'?

Reply:
https://cmake.org/cmake/help/v3.0/policy/CMP0002.html

Since `Logical target names must be globally unique.` I guess it would make sense to have it named differently.
There is a setting that would allow duplicates:
```
Custom targets must simply have globally unique names (unless one uses the global property ALLOW_DUPLICATE_CUSTOM_TARGETS with a Makefiles generator).
```
, but I'm unsure of the implications; maybe it would be best to have a clear distinction nevertheless.

Reply:
I wonder if https://cmake.org/cmake/help/latest/module/ExternalProject.html
may work around this in the nrn cmake.

Reply:
@alexsavulescu : in some external projects, which are typically used as submodule in other project, what I have seen is that the generic targets like `docs` are disabled if project is being built as submodule i.e.

```
if (NOT IS_USED_AS_SUBMODULE)
    add_custom_target(docs ...)
endif()
```

The reason is that, if project X is being used as submodule under Y then forcing project X to use different targets than project Y is annoying IMO. In this case we can do it because both are managed by us but not the case for other projects.

> I wonder if https://cmake.org/cmake/help/latest/module/ExternalProject.html

External project is much more pain and we are using it for MOD2c : https://github.com/BlueBrain/CoreNeuron/blob/b1ea553f19673ae0c107ed4c01c23c4484ef00c0/CMake/AddMod2cSubmodule.cmake#L37. I am going to remove that soon! :)

Reply:
Note that we are using this pattern already:

https://github.com/BlueBrain/nmodl/blob/f90f671e03fd537540de359e52e21852a335f2b1/CMakeLists.txt#L166
https://github.com/BlueBrain/CoreNeuron/blob/b1ea553f19673ae0c107ed4c01c23c4484ef00c0/CMakeLists.txt#L101

Reply:
Hmm... I wonder.... If I install sphinx and doxygen on this Mac, will I have the same issues when coreneuron is a submodule.  I expect so.

Reply:
I didn't understand. Do you mean for `docs` target if NEURON as well as CoreNEURON have the same target?
What I saw saying is that if coreneuron is submodule under NEURON then `docs` target of CoreNEURON won't be added in CMake.

Reply:
@pramodk `docs` is calling two other targets (`sphinx` and `doxygen`) that are created only if we have the respective tools installed. Those would clash as well since we have the same setup in `nrn` and `corenrn`

Reply:
Ok added the extra check. @nrnhines it should go fine with `nrn` now. Thanks @pramodk

Reply:
I'm a little confused about coreneuron as submodule docs. Wouldn't it be the case that the super module would want to install those docs. And similarly for neuron. If nrn is a submodule then docs could change to nrn_docs

Reply:
We can discuss how would like to see documentation of different projects : 1) whether all should be integrated into single tree or 2) there should be link to different projects doscs.

Obviously 1) has advantage of being able to search under single tree. But today this might be too early to do so? 

@alexsavulescu : another way to achieve previous use case would be:

```
if (CORENEURON_AS_SUBPROJECT)
  set(CORENEURON_TARGET_PREFIX "corenrn")
endif()

if(DOXYGEN_FOUND AND SPHINX_FOUND)
  add_custom_target(
    ${CORENEURON_TARGET_PREFIX}docs
   ...
   )
```
@nrnhines  : we can merge this PR as it is. But if we still thing we should re-enable docs with different target names then we can use above approach.

Reply:
I'm satisfied

Reply:
Option 1 is ideal and definitely the long term goal.

At this point I'd avoid doing more work here because we can end up with clashes in the `build/docs` folder as well, not just targets. My vision is as soon as we get production-ready tools to incorporate doxygen and jupyter notebooks directly into sphinx, we will have a singular target called `docs` in neuron.

Reply:
Agree ! all good then!